src/Framework/Telemetry/TelemetryItem.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Framework.Telemetry;
+
+internal record TelemetryItem(string Name, object Value, bool Hashed);
```

Comment:
The last boolean means that the value _will need to be hashed_, rather than _it is already hashed_ - correct?
```suggestion
internal record TelemetryItem(string Name, object Value, bool NeedsHashing);
```